spec/requests/alchemy/admin/pages_controller_spec.rb

```diff
@@ -641,93 +647,58 @@ module Alchemy
       end
 
       describe '#destroy' do
-        let(:clipboard) { session[:alchemy_clipboard] = {} }
+        let(:clipboard) { [{'id' => page.id.to_s}] }
         let(:page) { create(:alchemy_page, :public) }
 
-        before { clipboard['pages'] = [{'id' => page.id.to_s}] }
+        before do
+          allow_any_instance_of(described_class).to receive(:get_clipboard).with('pages') do
+            clipboard
+          end
+        end
 
         it "should also remove the page from clipboard" do
-          alchemy_xhr :post, :destroy, {id: page.id, _method: :delete}
-          expect(clipboard['pages']).to be_empty
+          xhr :delete, admin_page_path(page)
+          expect(clipboard).to be_empty
         end
       end
 
       describe '#publish' do
-        let(:page) do
-          stub_model Page,
-            published_at: nil,
-            public_on: nil,
-            name: "page",
-            parent_id: 1,
-            urlname: "page",
-            language: stub_model(Language),
-            page_layout: "bla"
-        end
-
-        before do
-          allow(@controller).to receive(:load_page).and_return(page)
-          @controller.instance_variable_set("@page", page)
-        end
+        let(:page) { create(:alchemy_page) }
 
         it "should publish the page" do
-          expect(page).to receive(:publish!)
-          alchemy_post :publish, { id: page.id }
+          expect {
+            post publish_admin_page_path(page)
+          }.to change { page.reload.published_at }
         end
```

Comment:
Feel free to actually directly cherry-pick that:

- https://github.com/mtomov/alchemy_cms/commit/8074d3c19b0a010866167025c876589e2f8aecea